File: No_Contracts/Server V1/src/main/java/com/rapid7/example/pact/server/demo/service/AnimalService.java
package com.rapid7.example.pact.server.demo.service;

import com.google.common.collect.Maps;
import com.rapid7.example.pact.server.demo.model.Animal;
import org.springframework.stereotype.Service;

import java.util.Collection;
import java.util.Map;

@Service
public class AnimalService {
    private Map<String, Animal> animalDB = Maps.newHashMap();

    public AnimalService () {
        animalDB.put("Tony", new Animal("Tony","Tiger","Orange"));
        animalDB.put("Orville", new Animal("Orville","Duck","Green"));
        animalDB.put("Elmo", new Animal("Elmo","Muppet","Red"));
        animalDB.put("Dory", new Animal("Dory","Fish","Blue"));
        animalDB.put("Ben", new Animal("Ben","Bear","Brown"));
        animalDB.put("Ed", new Animal("Ed","Horse","Black"));
        animalDB.put("Flipper", new Animal("Flipper","Dolphin","Grey"));
        animalDB.put("Ryan", new Animal("Ryan","Velociraptor","Ginger"));
    }

    public Collection<Animal> getAllAnimals() {
        return animalDB.values();
    }

    public Animal getAnimal(String name) {
        return animalDB.get(name);
    }

    public Animal addAnimal(Animal animal) {
        animalDB.put(animal.getName(), animal);
        return animal;
    }

    public void deleteAnimal(String name) {
        animalDB.remove(name);
    }
}

File: Server_Contracts/Server/api/services/AnimalService.js
'use strict';

let animalDbMap = new Map();
animalDbMap.set("Tony", {
    id:"Tony",
    species:"Tiger",
    color:"Orange"
});
animalDbMap.set("Orville", {
    id:"Orville",
    species:"Duck",
    color:"Green"
});
animalDbMap.set("Elmo", {
    id:"Elmo",
    species:"Muppet",
    color:"Red"
});
animalDbMap.set("Dory", {
    id:"Dory",
    species:"Fish",
    color:"Blue"
});
animalDbMap.set("Ben", {
    id:"Ben",
    species:"Bear",
    color:"Brown"
});
animalDbMap.set("Ed", {
    id:"Ed",
    species:"Horse",
    color:"Black"
});
animalDbMap.set("Flipper", {
    id:"Flipper",
    species:"Dolphin",
    color:"Grey"
});
animalDbMap.set("Ryan", {
    id:"Ryan",
    species:"Velociraptor",
    color:"Ginger"
});

function getAnimal(id) {
    const animal = animalDbMap.get(id);
    return animal;
}

function getAllAnimals() {

    let animalsList = [];
    for (let key of animalDbMap){
        animalsList.push(key[1])
    }

    return animalsList;
}

function addAnimal(animal) {
    animalDbMap.set(animal.id, animal);
}

function deleteAnimal(id) {
    animalDbMap.delete(id);
}

module.exports = {
    getAnimal,
    getAllAnimals,
    addAnimal,
    deleteAnimal,
};